src/NuGet.Core/NuGet.Protocol/LocalRepositories/LocalAutoCompleteResource.cs

```diff
@@ -108,8 +107,7 @@ public LocalAutoCompleteResource(FindLocalPackagesResource localResource)
                 return packages.Where(p => p.Identity.Version.ToString()
                     .StartsWith(versionPrefix, StringComparison.OrdinalIgnoreCase)
                     || p.Identity.Version.ToFullString().StartsWith(versionPrefix, StringComparison.OrdinalIgnoreCase))
-                    .Select(p => p.Identity.Version)
-                    .ToList();
+                    .Select(p => p.Identity.Version);
```

Comment:
Same as some of the above. 

This can be super problematic in an unobvious way because it can force multiple http/file reads if the enumeration is lazified.